packages/discovery-react-components/src/components/StructuredQuery/components/RemoveRuleRowButton/RemoveRuleRowButton.tsx

```diff
@@ -36,15 +36,20 @@ export const RemoveRuleRowButton: FC<RemoveRuleRowButtonProps> = ({
   setStructuredQuerySelection
 }) => {
   const handleOnClick = () => {
-    const filteredRow = structuredQuerySelection.groups[groupId].rows.filter(
+    const filteredGroupRows = structuredQuerySelection.groups[groupId].rows.filter(
       (row: number) => row !== rowId
     );
-    const isLastRuleInRuleGroup = filteredRow.length === 0;
+    const filteredRows = omit(structuredQuerySelection.rows, rowId);
+    const isLastRuleInRuleGroup = filteredGroupRows.length === 0;
     if (isLastRuleInRuleGroup) {
       setStructuredQuerySelection({
+        ...structuredQuerySelection,
         groups: {
           ...omit(structuredQuerySelection.groups, groupId)
         },
+        rows: {
+          ...filteredRows
+        },
         group_order: structuredQuerySelection.group_order.filter(id => id !== groupId)
```

Comment:
do you mind clarifying what the use of `group_order` is?

Reply:
Sure! See this discussion for the background: https://github.com/watson-developer-cloud/discovery-components/pull/42#discussion_r364437642. I can also add a comment clarifying if you think that would be helpful?